earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -433,17 +601,42 @@ DestroyDocument(khxml::DOMDocument *doc) throw()
     retval = true;
   } catch (...) {
   }
+  if (terminateCache)
+  {
+        khLockGuard guard(checkTermLock);
+        terminateGuard::instance().removeObj();
+        if (terminateGuard::instance().isLocked() &&
```

Comment:
Should this have a "not" in front of it - that is, do we want the instance to be locked or unlocked? Perhaps I don't understand the purpose of terminateGuard::instance().lock/unlock/isLocked.